src/arrow-util/src/builder.rs

```diff
@@ -129,72 +128,130 @@ impl ArrowBuilder {
-        ScalarType::Time => DataType::Time64(arrow::datatypes::TimeUnit::Microsecond),
-        ScalarType::Timestamp { .. } => {
-            DataType::Timestamp(arrow::datatypes::TimeUnit::Microsecond, None)
-        }
-        ScalarType::TimestampTz { .. } => DataType::Timestamp(
-            arrow::datatypes::TimeUnit::Microsecond,
-            // When appending values we always use UTC timestamps, and setting this to a non-empty
-            // value allows readers to know that tz-aware timestamps can be compared directly.
-            Some("+00:00".into()),
+        ScalarType::Time => (
+            DataType::Time64(arrow::datatypes::TimeUnit::Microsecond),
+            "time",
         ),
-        ScalarType::Bytes => DataType::LargeBinary,
+        ScalarType::Timestamp { .. } => (
+            DataType::Timestamp(arrow::datatypes::TimeUnit::Microsecond, None),
+            "timestamp",
+        ),
+        ScalarType::TimestampTz { .. } => (
+            DataType::Timestamp(
+                arrow::datatypes::TimeUnit::Microsecond,
+                // When appending values we always use UTC timestamps, and setting this to a non-empty
+                // value allows readers to know that tz-aware timestamps can be compared directly.
+                Some("+00:00".into()),
+            ),
+            "timestamptz",
+        ),
+        ScalarType::Bytes => (DataType::LargeBinary, "bytes"),
         ScalarType::Char { length } => {
             if length.map_or(false, |l| l.into_u32() < i32::MAX.unsigned_abs()) {
-                DataType::Utf8
+                (DataType::Utf8, "char")
             } else {
-                DataType::LargeUtf8
+                (DataType::LargeUtf8, "char")
             }
         }
         ScalarType::VarChar { max_length } => {
             if max_length.map_or(false, |l| l.into_u32() < i32::MAX.unsigned_abs()) {
-                DataType::Utf8
+                (DataType::Utf8, "varchar")
             } else {
-                DataType::LargeUtf8
+                (DataType::LargeUtf8, "varchar")
             }
         }
-        ScalarType::String => DataType::LargeUtf8,
-        ScalarType::MzTimestamp => {
-            // MZ timestamps are in milliseconds
-            DataType::Timestamp(arrow::datatypes::TimeUnit::Millisecond, None)
+        ScalarType::String => (DataType::LargeUtf8, "string"),
+        // Parquet does have a UUID 'Logical Type' in parquet format 2.4+, but there is no arrow
+        // UUID type, so we match the format (a 16-byte fixed-length binary array) ourselves.
+        ScalarType::Uuid => (DataType::FixedSizeBinary(16), "uuid"),
+        // Parquet does have a JSON 'Logical Type' in parquet format 2.4+, but there is no arrow
+        // JSON type, so for now we represent JSON as 'large' utf8-encoded strings.
+        ScalarType::Jsonb => (DataType::LargeUtf8, "jsonb"),
+        ScalarType::MzTimestamp => (DataType::UInt64, "mztimestamp"),
```

Comment:
tiniest nit: pretty sure `mz_timestamp` is the general form we use in docs, would be nice to match here

```suggestion
        ScalarType::MzTimestamp => (DataType::UInt64, "mz_timestamp"),
```